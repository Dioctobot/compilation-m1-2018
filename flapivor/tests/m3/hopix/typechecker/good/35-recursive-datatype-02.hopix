type T = N | `(T, T)

val leaf : T = N<>

val a_tree : T = `<> (`<> (`<> (leaf, leaf), `<>(leaf, `<>(leaf, leaf))), leaf)