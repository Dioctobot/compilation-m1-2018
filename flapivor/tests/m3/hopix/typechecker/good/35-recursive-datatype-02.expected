leaf : T
a_tree : T
